test/ao/sparsity/test_activation_sparsifier.py

```diff
@@ -78,6 +123,79 @@ def check_constructor(self, activation_sparsifier, model, defaults, sparse_confi
             # all the keys in combined_defaults should be present in sparsifier defaults
             assert config == combined_defaults.get(key, None)
 
+    def check_register_layer(self, activation_sparsifier, defaults, sparse_config, layer_args_list):
+        """Checks if layers in the model are correctly mapped to it's arguments.
+
+        Args:
+            activation_sparsifier (sparsifier object)
+                activation sparsifier object that is being tested.
+
+            defaults (Dict)
+                all default config (except sparse_config)
+
+            sparse_config (Dict)
+                default sparse config passed to the sparsifier
+
+            layer_args_list (list of tuples)
+                Each entry in the list corresponds to the layer arguments.
+                First entry in the tuple corresponds to all the arguments other than sparse_config
+                Second entry in the tuple corresponds to sparse_config
+        """
+        # check args
+        data_groups = activation_sparsifier.data_groups
+        assert len(data_groups) == len(layer_args_list)
+        for layer_args in layer_args_list:
+            layer_arg, sparse_config_layer = layer_args
+
+            # check sparse config
+            sparse_config_actual = copy.deepcopy(sparse_config)
+            sparse_config_actual.update(sparse_config_layer)
+
+            name = module_to_fqn(activation_sparsifier.model, layer_arg['layer'])
+            assert data_groups[name]['layer'] == layer_arg['layer']
```

Comment:
also I think ideally we should focus on testing the functionality instead of implementation details, it looks like we are testing implementation details here

Reply:
removed such tests